python/Ganga/GPIDev/Schema/Schema.py

```diff
@@ -243,14 +248,18 @@ def _getDefaultValueInternal(self, attr, val=None, check=False):
                 defvalue = _found_attrs[stored_attr_key]
             else:
                 defvalue = config[attr]
-                _found_attrs[stored_attr_key] = defvalue
+                from Ganga.GPIDev.Base.Proxy import stripProxy
+                ## Just in case a developer puts the proxied object into the default value!
+                _found_attrs[stored_attr_key] = stripProxy(defvalue)
```

Comment:
If a developer adds a possible type of ```Proxy.File``` class and then sets a value of ```File``` in some part of their code the TypeCheck will fail. (```File``` is not an instance of ```Proxy.File```)
(It used to simply ignore this and assume as you're setting an object which looks like it's a Ganga class it's probably OK. I'm happy to revert this functionality but this is adding inconsistent behaviour into TypeCheck imo. If we know it's a Ganga class we stop caring about checking the type?...
).
I've added a better checking mechanism into TypeCheck which checks if the Ganga object is an instance of the class but raised at least one issue so I was wanting to be explicit here.